Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -0,0 +1,2212 @@
+    errss << "One and only one in 'PeakCenters' (vector) and "
+             "'PeakCentersWorkspace' shall be given. "
+          << "'PeakCenters' has size " << m_peakCenters.size()
+          << ", and name of peak center workspace "
+          << "is " << peakpswsname;
+    throw std::invalid_argument(errss.str());
+  }
+
+  return;
+}
+
+//----------------------------------------------------------------------------------------------
+/** Processing peak fitting tolerance information from input.  The parameters
+ * that are
+ * set including
+ * 2. m_peakPosTolerances (vector)
+ * @brief FitPeaks::ProcessInputPeakTolerance
+ */
+void FitPeaks::ProcessInputPeakTolerance() {
+  // check code integrity
+  if (m_numPeaksToFit == 0)
+    throw std::runtime_error("ProcessInputPeakTolerance() must be called after "
+                             "ProcessInputPeakCenters()");
+
+  // peak tolerance
+  m_peakPosTolerances = getProperty("PositionTolerance");
+
+  if (m_peakPosTolerances.size() == 0) {
+    // case 2, 3, 4
+    m_peakPosTolerances.clear();
+    m_peakPosTolCase234 = true;
+  } else if (m_peakPosTolerances.size() == 1) {
+    // only 1 uniform peak position tolerance is defined: expand to all peaks
+    double peak_tol = m_peakPosTolerances[0];
+    m_peakPosTolerances.resize(m_numPeaksToFit, peak_tol);
+  } else if (m_peakPosTolerances.size() != m_numPeaksToFit) {
+    // not uniform but number of peaks does not match
+    g_log.error() << "number of peak position tolerance "
+                  << m_peakPosTolerances.size()
+                  << " is not same as number of peaks " << m_numPeaksToFit
+                  << "\n";
+    throw std::runtime_error("Number of peak position tolerances and number of "
+                             "peaks to fit are inconsistent.");
+  }
+
+  // minimum peak height: set default to zero
+  m_minPeakHeight = getProperty("MinimumPeakHeight");
+  if (isEmpty(m_minPeakHeight))
+    m_minPeakHeight = 0.;
+
+  return;
+}
+
+//----------------------------------------------------------------------------------------------
+/** Convert the input initial parameter name/value to parameter index/value for
+ * faster access
+ * according to the parameter name and peak profile function
+ * @brief FitPeaks::ConvertParametersNameToIndex
+ * Output: m_initParamIndexes will be set up
+ */
+void FitPeaks::ConvertParametersNameToIndex() {
+  // get a map for peak profile parameter name and parameter index
+  std::map<std::string, size_t> parname_index_map;
+  for (size_t iparam = 0; iparam < m_peakFunction->nParams(); ++iparam)
+    parname_index_map.insert(
+        std::make_pair(m_peakFunction->parameterName(iparam), iparam));
+
+  // define peak parameter names (class variable) if using table
+  if (m_profileStartingValueTable)
+    m_peakParamNames = m_profileStartingValueTable->getColumnNames();
+
+  // map the input parameter names to parameter indexes
+  for (size_t i = 0; i < m_peakParamNames.size(); ++i) {
```

Comment:
`FitPeaks.cpp:649:3: warning: use range-based for loop instead [modernize-loop-convert]`